lib/ruby_lsp/handler.rb

```diff
@@ -125,7 +125,7 @@ def respond_with_selection_ranges(uri, positions)
 
     def respond_with_semantic_highlighting(uri)
       store.cache_fetch(uri, :semantic_highlighting) do |document|
-        Requests::SemanticHighlighting.run(document)
+        Requests::SemanticHighlighting.new(document, encoder: Requests::Support::SemanticTokenEncoder).run
```

Comment:
Now that we've decoupled the Encoder class from the Request class, we can start sending in an "instance" of the encoder instead of the class of it:
```suggestion
        Requests::SemanticHighlighting.new(document, encoder: Requests::Support::SemanticTokenEncoder.new).run
```